ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageManager.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreException;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The storage manager that does book keeping for all of the partitions handled by this vcr node.
+ */
+public class CloudStorageManager implements StoreManager {
+  private final ConcurrentMap<PartitionId, CloudBlobStore> partitionToStore;
+  private final VerifiableProperties properties;
+  private final CloudDestination cloudDestination;
+  private final VcrMetrics vcrMetrics;
+  private final ClusterMap clusterMap;
+  private static final Logger logger = LoggerFactory.getLogger(CloudStorageManager.class);
+
+  public CloudStorageManager(VerifiableProperties properties, VcrMetrics vcrMetrics, CloudDestination cloudDestination,
+      ClusterMap clusterMap) {
+    partitionToStore = new ConcurrentHashMap<>();
+    this.properties = properties;
+    this.cloudDestination = cloudDestination;
+    this.vcrMetrics = vcrMetrics;
+    this.clusterMap = clusterMap;
+  }
+
+  @Override
+  public boolean addBlobStore(ReplicaId replica) {
+    CloudBlobStore cloudBlobStore = partitionToStore.computeIfAbsent(replica.getPartitionId(),
+        store -> new CloudBlobStore(properties, replica.getPartitionId(), cloudDestination, clusterMap, vcrMetrics));
+    partitionToStore.put(replica.getPartitionId(), cloudBlobStore);
```

Comment:
I think that computeIfAbsent will already insert the entry into the map after computing so this call might not be needed

Reply:
fixed. also all tests passing.